hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -196,6 +196,11 @@ public class HoodieTableConfig extends HoodieConfig {
       .withDocumentation("If true, partition metafiles are saved in the same format as basefiles for this dataset (e.g. Parquet / ORC). "
           + "If false (default) partition metafiles are saved as properties files.");
 
+  public static final ConfigProperty<Boolean> DROP_PARTITION_COLUMNS = ConfigProperty
+      .key("hoodie.datasource.write.drop.partition.columns")
+      .defaultValue(false)
+      .withDocumentation("When set to true, will not write the partition columns into hudi. By default, false.");
```

Comment:
Why we need this option, is this a promotion ?

Reply:
> Why we need this option, is this an improvement ?

This is not a new config. It was added long back. This PR is to fix some behaviors when this is enabled